models/comment.rb

```diff
@@ -31,17 +32,21 @@ class Comment < Content
   index({_type: 1, comment_thread_id: 1, author_id: 1, updated_at: 1})
   index({comment_thread_id: 1, author_id: 1, created_at: 1})
 
-  index_name Content::ES_INDEX_NAME
-
-  mapping do
-    indexes :body, type: :string, analyzer: :english, stored: true, term_vector: :with_positions_offsets
-    indexes :course_id, type: :string, index: :not_analyzed, included_in_all: false
-    indexes :comment_thread_id, type: :string, index: :not_analyzed, included_in_all: false, as: 'comment_thread_id'
-    indexes :commentable_id, type: :string, index: :not_analyzed, included_in_all: false, as: 'commentable_id'
-    indexes :group_id, type: :string, index: :not_analyzed, included_in_all: false, as: 'group_id'
-    indexes :context, type: :string, index: :not_analyzed, included_in_all: false, as: 'context'
-    indexes :created_at, type: :date, included_in_all: false
-    indexes :updated_at, type: :date, included_in_all: false
+  index_name = "comment"
+
+  mapping dynamic: 'false' do
+    indexes :body, type: :text, store: true, term_vector: :with_positions_offsets
+    indexes :course_id, type: :keyword
+    indexes :comment_thread_id, type: :keyword
+    indexes :commentable_id, type: :keyword
+    indexes :group_id, type: :keyword
+    indexes :context, type: :keyword
+    indexes :created_at, type: :date
+    indexes :updated_at, type: :date
+  end
+
+  def as_indexed_json(options={})
```

Comment:
Hmm... Mutable default function parameter?

Reply:
[off doc link](https://www.rubydoc.info/gems/elasticsearch-model/Elasticsearch%2FModel%2FSerializing%2FInstanceMethods:as_indexed_json). I think that can use like is written in this doc.